### ISSUE TITLE
Using linear or log spaced layers grid for vertical sampling changes the final model

### ISSUE THREAD

**Issue by admasson** [ladsantos/p-winds#37]
Hi Leonardo,

First of all, thank you very much for your work in providing this code to the community in a publicly available way :) !

I'm using p-winds for He triplet characterization in some high-resolution spectroscopic data in the nIR.
I noticed that using a linearly spaced or log-spaced grid for the vertical sampling of the layers will change the final model output despite using the same other parameters as input.

For example, I took the first cells from the advanced tutorial on p-winds documentation (cells [1] to [6] here: https://p-winds.readthedocs.io/en/latest/advanced_tutorial.html) and tested by only changing the following line in cell [3]:

> r = np.logspace(0, np.log10(20), 100)

to

> r = np.linspace(1,20, 100)

I get the following output for cell [6]:
![aze](https://user-images.githubusercontent.com/19845488/219428017-e07ba5be-9f6b-440f-9772-52d4bf1ccb8a.png)

This seems to be linked with the resolution of the sampling, as increasing the number of layers from 100 to 1000 in the linear case gives an output model closer to the log case with 100 layers.

Thus, my question is: is there a physical or computational reason behind the choice of a log grid by default in the example?  Was the p-winds code optimized for the use of a log-spaced grid for example?

**Comment by ladsantos** [comment]
Hi. 

TL;DR: Using log-space altitudes ensures that the code properly calculates the radial profiles of metastable He while keeping the computational cost as low as possible.

So, the reason why I recommend using log-space is mainly because the structure of the upper atmosphere varies significantly with altitude. In addition, the shape of the theoretical transmission spectrum is strongly dependent on the He densities near the planet's surface, so more accurate results are obtained when you properly sample the layers closer to the surface.

For instance, if you use `r = np.linspace(1, 20, 100)`, here's how the He distribution looks like:
<img width="594" alt="Screen Shot 2023-02-23 at 11 49 07 AM" src="https://user-images.githubusercontent.com/2818188/220975388-72a11855-0fa9-461c-9a6a-b28b8c56757a.png">

But if you use `r = np.logspace(0, np.log10(20), 100)`, here's how it looks:
<img width="603" alt="Screen Shot 2023-02-23 at 11 49 30 AM" src="https://user-images.githubusercontent.com/2818188/220975526-1ede28b8-5f0f-4567-ba58-0a4ce0c040ea.png">

See, if you use linear space, the resulting He profiles do not properly sample the inner layers, and you miss a significant "chunk" of the metastable He precisely in the altitudes where they are more important. This is the cause of the discrepancy in the theoretical transmission spectra you obtained. If you really want to stick to linear spacing, one way of getting around this problem is by sampling the altitudes more finely. This is the structure I get when I use `r = np.linspace(1, 20, 500)`:

<img width="600" alt="Screen Shot 2023-02-23 at 11 57 29 AM" src="https://user-images.githubusercontent.com/2818188/220976607-146c1280-54b3-422f-8788-2c538deac984.png">

But this is more computationally costly, because we are calculating profiles for 500 points instead of 100.

**Comment by admasson** [comment]
Ok, this clears things up. Thank you :) !
